src/log/log.cr

```diff
@@ -65,14 +74,15 @@ class Log
       dsl = Emitter.new(@source, severity, exception)
       result = yield dsl
 
-      case result
-      when Entry
-        backend.dispatch result
-      when Nil
-        # emit nothing
-      else
-        backend.dispatch dsl.emit(result.to_s)
-      end
+      entry =
+         case result
+         when Entry
+           result
+         else
+           dsl.emit(result.to_s)
+         end
```

Comment:
I think we should modify this to handle both use cases. I.e. if `result.nil?` don't log anything _only_ if `exception` is also `nil`.

Reply:
I had moved that logic to line 85, that is to only dispatch the log entry unless both `exception` and the block `result` are `nil`, because I thought it was easier to understand. Do you want that check done earlier, when dealing with the block result in the case statement?

Reply:
Oh nice, I don't have strong feelings either way so that works for me!

Reply:
But now `entry` is always generated, yet can be thrown away immediately. It should be:

```crystal
unless exception.nil? && result.nil?
  entry = ...
  backend.dispatch entry
end
```

Reply:
@ysbaddaden yes good point, I've now updated the logic to only create the entry if it's going to be dispatched like you suggested. How does it look now?

Reply:
@ysbaddaden I also just changed the `unless exception.nil? && result.nil?` check to be `unless result.nil? &&  exception.nil?` which I assume will let the check be short circuited when `results.nil?` is `false` which would be the majority of the time. Do you agree?